gov: pools gov prop supercharge liquidity

czarcas7ic opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
v    If your PR doesn't close an issue, that's OK!  Just remove the Closes: #XXX line!
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

Closes: #5307 

## What is the purpose of the change

We currently only allow for a single pool to be added at a time via gov prop. This PR allows for multiple pools to be created with a single gov message.

## Testing and Verifying

Tests modified to expect a list of poolRecords rather than just one

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

AlpinYukseloglu commented on proto/osmosis/concentrated-liquidity/gov.proto:
```diff
@@ -5,30 +5,18 @@ import "gogoproto/gogo.proto";
-    (gogoproto.moretags) = "yaml:\"spread_factor\"",
-    (gogoproto.nullable) = false
-  ];
+  repeated PoolRecord pool_records = 3 [ (gogoproto.nullable) = false ];
```
Should we be adding a tag here? e.g  `(gogoproto.moretags) = "yaml:\"pool_records\""`

AlpinYukseloglu commented on x/concentrated-liquidity/types/gov.go:
```diff
@@ -15,71 +15,84 @@ const (
-	return nil
+	var b strings.Builder
+	b.WriteString(fmt.Sprintf(`Create Concentrated Liquidity Pool Proposal:
+  Title:                 %s
```
nit: this formatting looks off

AlpinYukseloglu commented on x/concentrated-liquidity/types/gov_test.go:
```diff
@@ -10,39 +10,52 @@ import (
+		{
+			Denom0:             "uion",
+			Denom1:             "uosmo",
+			TickSpacing:        100,
```
nit: we should have different tick spacings here as well & if possible another set of records (non-blocking)

AlpinYukseloglu reviewed (approved):
LGTM, nice work. Left a couple nits but approving since none are blocking. Note that I only reviewed diff & didn't test new cli command on localosmosis

czarcas7ic commented:
@AlpinYukseloglu addressed all comments in [this commit](https://github.com/osmosis-labs/osmosis/pull/5345/commits/7a15c33e8f7d5da972bb9b101756dbdf8f6bc1d2), thanks!
